packages/datadog-plugin-pino/src/index.js

```diff
@@ -19,15 +19,21 @@ function createWrapGenLog (tracer, config) {
 
       return function logWithTrace (a, b, c, d, e, f, g, h, i, j, k) {
         const args = [a, b, c, d, e, f, g, h, i, j, k]
+        let argsLength = arguments.length
```

Comment:
Instead of keeping track of the length, would it be possible to use `args = [a, b, etc].slice(arguments.length)`? Then you would get a new array with the correct length upfront, so no need for the increment nor the loop.

Reply:
That sounds good to me 👍